DQM/BeamMonitor/plugins/BeamSpotDipServer.cc

```diff
@@ -0,0 +1,740 @@
+  }
+
+  return outstr;
+}
+
+/*****************************************************************************/ 
+void BeamSpotDipServer::problem()
+{
+  if(verbose) cout << "[Dip] no update | alive = " << alive << endl;
+
+  lsCount++;
+
+  if ((lsCount % timeoutLS[0] == 0)
+   && (lsCount % timeoutLS[1] != 0)) // first time out
+  {
+    if (!alive.test(1)) alive.flip(1);
+    if (!alive.test(2))
+    {
+      if (!alive.test(7)) fakeRcd();
+                     else trueRcd();
+
+      stringstream warnMsg;
+      warnMsg << "No new data for " << lsCount << " LS";
+      publishRcd("Uncertain",warnMsg.str(),false,false);
+    } else {
+      fakeRcd();
+
+      stringstream warnMsg;
+      warnMsg << "No new data for " << lsCount << " LS: "
+               << tkStatus();
+      publishRcd("Bad",warnMsg.str(),false,false);
+    }
+  }
+  else if (lsCount % timeoutLS[1] == 0) // second time out
+  {
+    if (!alive.test(2)) alive.flip(2);
+    fakeRcd();
+
+    stringstream warnMsg;
+    warnMsg << "No new data for " << lsCount << " LS: " << tkStatus();
+    publishRcd("Bad",warnMsg.str(),false,false);
+  }
+}
+
+/*****************************************************************************/ 
+bool BeamSpotDipServer::readRcd(const BeamSpotOnlineObjects & bs)
+// read from database
+{
+  runnum = bs.GetLastAnalyzedRun();
+
+  startTime      = bs.GetCreationTime();
+  startTimeStamp = bs.GetCreationTime(); // not provided by BeamSpotOnlineObject
+    endTime      = bs.GetCreationTime();
+    endTimeStamp = bs.GetCreationTime(); // not provided by BeamSpotOnlineObject
+
+  lumiRange = to_string(bs.GetLastAnalyzedLumi());
+  currentLS = bs.GetLastAnalyzedLumi();
+
+  type = bs.GetBeamType();
+
+  if(verbose)
+    cout << "[Dip] run: " << runnum
+         << ", LS: "    << currentLS 
+         << ", time: " << startTime << " " << startTimeStamp
+         << ", type: " << type << endl;
+
+  if (testing)          quality = qualities[0]; // Uncertain
+  else if (type >= 2)   quality = qualities[2]; // Good
+                   else quality = qualities[1]; // Bad
+
+  x = bs.GetX();
+  y = bs.GetY();
+  z = bs.GetZ();
+
+  sigma_z = bs.GetSigmaZ();
+  dxdz    = bs.Getdxdz(); 
+  dydz    = bs.Getdydz();
+  width_x = bs.GetBeamWidthX();
+  width_y = bs.GetBeamWidthX();
+
+  err_x       = bs.GetXError();
+  err_y       = bs.GetYError();
+  err_z       = bs.GetZError();
+  err_sigma_z = bs.GetSigmaZError();
+  err_dxdz    = bs.GetdxdzError();
+  err_dydz    = bs.GetdydzError();
+  err_width_x = bs.GetBeamWidthXError();
+  err_width_y = bs.GetBeamWidthYError();
+
+      events = 0.; // not provided by BeamSpotOnlineObject
+      meanPV = 0.; // not provided by BeamSpotOnlineObject
+  err_meanPV = 0.; // not provided by BeamSpotOnlineObject
+       rmsPV = 0.; // not provided by BeamSpotOnlineObject
+   err_rmsPV = 0.; // not provided by BeamSpotOnlineObject
+       maxPV = 0.; // not provided by BeamSpotOnlineObject
```

Comment:
@sikler could you provide:
- a brief description of these parameters
- an example of the old data from DIP (just to see these parameters)
- the code where they were read in the old script?
Again these can maybe be added to the BeamSpotOnlineObject.

Reply:
They refer to the number of primary vertices per event.
Some example rounded numbers:
meanPV 3.4
meanErrPV 0.2
rmsPV 0.5
rmsErrPV 0.1
maxPV 100
nPV 2000
where mean of the number of PVs, uncertainty of that mean, rms, uncertainty of rms, maximum of the number of PVs, and number of PVs altogether analyzed in the lumisection in question are provided.